integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientExternalIdTest.java

```diff
@@ -0,0 +1,199 @@
+
+        // Reactivate Client action
+        jsonPayload = clientHelper.getReactivateClientAsJSON();
+        commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec, clientExternalId,
+                ClientHelper.REACTIVATE_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        log.info("Client data id {} and external Id {}", commandResponse.getResourceId(), clientExternalId);
+        assertEquals(clientId.intValue(), commandResponse.getResourceId());
+
+        clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.PENDING, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        // Reject Client action
+        jsonPayload = clientHelper.getRejectClientAsJSON();
+        commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec, clientExternalId,
+                ClientHelper.REJECT_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        log.info("Client data id {} and external Id {}", commandResponse.getResourceId(), clientExternalId);
+        assertEquals(clientId.intValue(), commandResponse.getResourceId());
+
+        clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.REJECTED, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        // Activate Client action
+        jsonPayload = ClientHelper.getActivateClientAsJSON(ClientHelper.DEFAULT_DATE);
+        commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec, clientExternalId,
+                ClientHelper.ACTIVATE_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        log.info("Client data id {} and external Id {}", commandResponse.getResourceId(), clientExternalId);
+        assertEquals(clientId.intValue(), commandResponse.getResourceId());
+
+        clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.ACTIVE, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        GlobalConfigurationHelper.manageConfigurations(requestSpec, responseSpec,
+                GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, false);
+    }
+
+    private void fetchClientByExternalId(final String externalId) {
+        GetClientsClientIdResponse clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, externalId);
+        assertNotNull(clientResponse);
+        assertEquals(externalId, clientResponse.getExternalId());
+    }
+}
```

Comment:
Missing test cases:
- Retrieve all client and client details contains the external id
- Update client by external id and result contains the externalId
- Delete client by external id
- Do actions on client by external id (activate, close, reactive, undo, ...) all these actions are retrieving the externalId in the result
- Fetch client account by externalId
- Fetch obligee details by externalId
- Fetch transfer proposal date by externalId

Reply:
Done!